src/internal/dbutil/tx.go

```diff
@@ -113,7 +114,7 @@ func WithBackOff(bo backoff.BackOff) WithTxOption {
 // WithTx calls cb with a transaction,
 // The transaction is committed IFF cb returns nil.
 // If cb returns an error the transaction is rolled back.
-func WithTx(ctx context.Context, db *pachsql.DB, cb func(tx *pachsql.Tx) error, opts ...WithTxOption) error {
+func WithTx(ctx context.Context, db *pachsql.DB, cb func(ctx context.Context, tx *pachsql.Tx) error, opts ...WithTxOption) error {
```

Comment:
I would name the inner ctx something else just to reduce ambiguity
```suggestion
func WithTx(ctx context.Context, db *pachsql.DB, cb func(ctx2 context.Context, tx *pachsql.Tx) error, opts ...WithTxOption) error {
```